xarray/core/combine.py

```diff
@@ -207,7 +212,10 @@ def _dataset_concat(datasets, dim, data_vars, coords, compat, positions):
 
     dim, coord = _calc_concat_dim_coord(dim)
     datasets = [as_dataset(ds) for ds in datasets]
-    datasets = align(*datasets, join='outer', copy=False, exclude=[dim])
+    if not prealigned:
+        datasets = align(*datasets, join='outer', copy=False, exclude=[dim])
+    else:
+        coords = 'minimal'
```

Comment:
What about just adding the option `coords='prealigned'`?

My initial thought was that, for prealigned data, all coords should just be drawn from the first object. But on second thought, what if there are other coords in the later dataset that do need to be concatenated, e.g. concat over `time` with an auxiliary coordinates `iteration_number` with dimension `time`.

It definitely doesn't work with `coords='different'`. I have not tried all the other options. I have a hard time conceptualizing what the different `coords` options do. Some guidance would be very welcome. I don't really understand what the function `_calc_concat_over` does.